packages/react-components/src/overlay/src/usePopup.ts

```diff
@@ -1,13 +1,35 @@
+import { AutoFocusChildOptions, mergeProps, useAutoFocusChild, useCommittedRef, useControllableState, useEventCallback, useFocusManager, useFocusScope, useId, useMergedRefs } from "../../shared";
+import { FocusEvent, SyntheticEvent, useCallback, useState } from "react";
+import { Placement } from "@popperjs/core";
 import { isNil, isNumber } from "lodash";
 import { isTargetParent } from "./isTargetParent";
-import { mergeProps, useAutoFocusChild, useCommittedRef, useControllableState, useEventCallback, useFocusManager, useFocusScope, useId, useMergedRefs } from "../../shared";
-import { useCallback, useState } from "react";
 import { useOverlayLightDismiss } from "./useOverlayLightDismiss";
 import { useOverlayPosition } from "./useOverlayPosition";
 import { useOverlayTrigger } from "./useOverlayTrigger";
 import { useRestoreFocus } from "./useRestoreFocus";
 
-export function usePopup(type, {
+export interface UsePopupProps {
+    id?: string;
+    open?: boolean;
+    defaultOpen?: boolean;
+    onOpenChange?: (event: SyntheticEvent, newValue: boolean) => void;
+    hideOnEscape?: boolean;
+    hideOnLeave?: boolean;
+    hideOnOutsideClick?: boolean;
+    restoreFocus?: boolean;
+    autoFocus?: boolean | number;
+    autoFocusOptions?: AutoFocusChildOptions;
+    trigger?: "click" | "hover";
+    hasArrow?: boolean;
+    position?: Placement;
+    offset?: number[];
+    allowFlip?: boolean;
+    allowPreventOverflow?: boolean;
+    boundaryElement?: HTMLElement,
+    keyProp?: string;
+}
+
+export function usePopup(type: boolean | "menu" | "listbox" | "tree" | "grid" | "dialog", {
```

Comment:
I modified the type as you asked. And why boolean? because i copied the type from the generated aria-haspopup type from typescript
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L1630